cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -106,6 +106,13 @@ func (a *StaticAutoscaler) cleanUpIfRequired() {
 
 // RunOnce iterates over node groups and scales them up/down if necessary
 func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError {
+	defer func() {
```

Comment:
This uses CSR, shouldn't we move it after updateClusterState, similar to when we defer writing status configmap? Otherwise you can't really assume anything you get from CSR is true.